### PR TITLE
Hello rails

### PR DESCRIPTION
**Pull request by Kandy-Peter**
### Rails
    - Create a new Rails app called 'hello-rails-react'.
    - Initialize your project with Git.
    - Set up your project with webpacker the react-rails gems as describe in the previous lesson's tutorial.
    - Make sure that your project has a Postgres database set up.
    - Create a table for storing your messages and create 5 different greetings.
    - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
   - Create a static view that will be the root of your app.
  
### React

    - Create the App component with react-router and render it in your static view.
    - Create the Greeting component that will display a greeting. Set it up as a route in your App component.
    - Create a store, an action and a reducer that will connect to the API endpoint to get the random greeting.
    - Display the random `greeting` in your Greeting component.

### CODE REVIEW

File: app/javascript/redux/greeting.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      default:
+        return state;
+    };
+};
+
+const fetchGreetings = () => async (dispatch) => {
+  const result = await axios.get('/api/v1/greetings');
+    const data = result.data.greetings
+    dispatch(getGreeting(data));
+};
```
Comment (Gambit142) [inline app/javascript/redux/greeting.js]: - [ ] Nice work setting up this application. I would like to point to you that the data you get from the API endpoint is an object that includes a key `data` which is an array that has an object This object in the array has a key tet which contains the greetings. Kindly retrieve the right data. 👍 

`const data = result.data[0].text`

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  return (
+    <p>Greeting: {text}</p>
+    );
+  };
+
+  Greeting.propTypes = {
+    greeting: PropTypes.string
+  };
+  export default Greeting;
```
Comment (Gambit142) [inline app/javascript/components/Greeting.js]: - [ ] When I route to the `Greeting` component, I do not get any greetings. I get an empty page each time I refresh the page. Kindly check the way you configured your reducer and your store in your redux files. 

_**Hint:**_ In your `store.js`, the way you are importing the reducer is wrong. Also, in the `Greeting` component, you want to access `state.reducer.greeting` when accessing the store using `useSelector`.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Kandy-Peter 👋, 

Good job so far!

## Highlights
- Installed `webpacker` and `react-rails` gems 💯 
- Set up the MVC on serverside properly 💯 
- Installed react-router-dom and set up routes correctly 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by uwadonat — APPROVED** [review]
## Hi @Kandy-Peter,

Thank you for implementing all changes suggested by previous reviewer. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
cong


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**